### DISCUSSION TITLE
Add Autoscaling

### DISCUSSION THREAD

**Discussion by mithleshjs** [planetscale/discussion#539] [Feature Requests] [answered]
Before the recent price change, Scaler Plan used to have autoscaling support which have now been limited to Enterprise plan only. In Scalar Pro plans we are made to select CPU/RAM which somehow defeats the purpose of using PlanetScale as one has to keep eyes on the database. There could be traffic spike during odd hours or at any other unspecified time. The DB should be able to handle all such scenerios.

**Comment by orware** [comment, marked as answer]
Hello @mithleshjs,

We appreciate the input!

Storage does autoscale for all databases which is important to mention, but other resources such as CPU / RAM don't currently autoscale, which is why the Scaler Pro plans were introduced to allow users to have more awareness and control over the resources directly assigned to their important production branches.

In the future we can continue to expand on what is possible with Scaler Pro that may include options to allow for instance size increases automatically perhaps, but there isn't anything official that can be shared about what may arrive in the future at this time.

I definitely hear where you are coming from with your last few sentences, and while that's an ideal goal for us to aim for, for now it's important to make sure users right-size the database workload with an appropriate amount of resources and monitor CPU / RAM usage to know when it may be time to increase (or even decrease) the amount of resources that may be needed, which is a lot easier to adjust now with our Scaler Pro offerings.

**Reply by etc-tiago** [reply]
It makes no sense.
The vast majority of applications have partial activities 24 hours a day.
In my case, from 8 am to 6 pm, the use is intense.
From 6pm to 10pm usage is reduced and from 10pm to 8am it is almost non-existent.

Should I keep reminding myself and repeating myself to avoid unnecessary expenses?

**Reply by ayrton** [reply]
@etc-tiago you can programmatically change your cluster size via our API https://planetscale.com/changelog/branch-cluster-api-endpoint

**Reply by etc-tiago** [reply]
@ayrton This is pretty cool, but can I see what the percentage of usage is?
If not, I will still have to go into the panel to check if the change is necessary.

What would be interesting would be an endpoint that returns at least 3 results, interval (for example 5 minutes), minimum and maximum usage for cpu and memory...

**Comment by dellow** [comment]
In all honestly I was surprised by this also. I mean, even the name "PlanetScale" would indicate that's kind of the point of the product.

It's probably worthwhile allowing users to select and fix resources if that's how they want to do it but I would expect autoscaling to be synonymous with PlanetScale and was disappointed when I realised it simply wasn't.

**Comment by Zay-Vidcade** [comment]
Same here. I started using PS because of autoscaling. We are Application engineers. We want to focus on the software part and leave the database part to you (an Pay you). Looks like with the introduction of your paid plan, that seems not to be the case. 

I can see myself switching to another startups that offers autoscaling at an affordable cost. The current model is not "giving" for me.
